File: Bronze/2021 January - Bronze/UdderedButNotHerd.cpp
#include <bits/stdc++.h>

using namespace std;

string alph, word;
int times = 0;
vector<int> hello;

int main() {
    ios::sync_with_stdio(0); cin.tie(0);

    cin >> alph >> word;

    int counter = 0;
    while(counter < word.size()) {
        char letter = word[counter];

        for(int i = 0; i < alph.size(); i++) {
            if(counter >= word.size()) break;

            if(alph[i] == letter) {
                counter++;
                letter = word[counter];
            }
        }
        
        times++;
    }

    cout << times;
}

File: Bronze/2021 January - Bronze/UdderedButNotHerd.java
import java.util.*;
import java.io.*;

public class UdderedButNotHerd {

    public static void main(String[] args) throws IOException {
        BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));

        String alph = sc.readLine();
        String word = sc.readLine();
        int times = 0;

        int counter = 0;
        while(counter < word.length()) {
            char letter = word.charAt(counter);

            for(int i =0 ; i < alph.length(); i++) {
                if(counter >= word.length()) break;

                if(alph.charAt(i) == letter) {
                    counter++;
                    if(counter >= word.length()) break;
                    letter = word.charAt(counter);
                }
            }

            times++;
    
        }

        System.out.println(times);
    }
}
    

